-Ainfer causes crash for basic assignment

When using `-Ainfer` with a basic assignment, I get a crash. Here's the code
```java
import org.checkerframework.checker.nullness.qual.*;

public class C {
    public void f(Object param) {
        param = null;
    }
}
```
```
$CHECKERFRAMEWORK/checker/bin/javac C.java -processor Nullnesschecker -Ainfer=stubs
```
```
error: Tried printing an unprintable class to a stub file during WPI: C
  ; The Checker Framework crashed.  Please report the crash.
  Compilation unit: C.java
  Exception: java.lang.Throwable; java.lang.Throwable
  	at org.checkerframework.javacutil.BugInCF.<init>(BugInCF.java:14)
  	at org.checkerframework.common.wholeprograminference.SceneToStubWriter.isPrintable(SceneToStubWriter.java:646)
  	at org.checkerframework.common.wholeprograminference.SceneToStubWriter.writeImpl(SceneToStubWriter.java:597)
  	at org.checkerframework.common.wholeprograminference.SceneToStubWriter.write(SceneToStubWriter.java:89)
  	at org.checkerframework.common.wholeprograminference.scenelib.ASceneWrapper.writeToFile(ASceneWrapper.java:146)
  	at org.checkerframework.common.wholeprograminference.WholeProgramInferenceScenesStorage.writeScenes(WholeProgramInferenceScenesStorage.java:110)
  	at org.checkerframework.common.wholeprograminference.WholeProgramInferenceScenes.writeResultsToFile(WholeProgramInferenceScenes.java:423)
  	at org.checkerframework.framework.type.AnnotatedTypeFactory.postProcessClassTree(AnnotatedTypeFactory.java:1121)
  	at org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:327)
  	at org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:165)
  	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:808)
  	at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:56)
  	at org.checkerframework.framework.source.SourceVisitor.visit(SourceVisitor.java:82)
  	at org.checkerframework.framework.source.SourceChecker.typeProcess(SourceChecker.java:894)
  	at org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:510)
  	at org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:190)
  	at jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:828)
  	at jdk.compiler/com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:132)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1418)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1375)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:973)
  	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:311)
  	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:170)
  	at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:57)
  	at jdk.compiler/com.sun.tools.javac.Main.main(Main.java:43)
C.java:5: error: [assignment.type.incompatible] incompatible types in assignment.
        param = null;
                ^
  found   : null
  required: @Initialized @NonNull Object
2 errors
```
This does not occur with `-Ainfer=jaifs`.
Like with #3422 I get a different crash with `checker/bin-devel/javac`,
```
$CHECKERFRAMEWORK/checker/bin-devel/javac C.java -processor Nullnesschecker -Ainfer=stubs
```
```
error: SourceChecker.typeProcess: unexpected Throwable (NoSuchMethodError) while processing C.java; message: 'java.lang.String scenelib.annotations.util.JVMNames.getJVMMethodSignature(javax.lang.model.element.ExecutableElement)'
  ; The Checker Framework crashed.  Please report the crash.
  Compilation unit: C.java
  Last visited tree at line 3 column 1:
  public class C {
  Exception: java.lang.NoSuchMethodError: 'java.lang.String scenelib.annotations.util.JVMNames.getJVMMethodSignature(javax.lang.model.element.ExecutableElement)'; java.lang.NoSuchMethodError: 'java.lang.String scenelib.annotations.util.JVMNames.getJVMMethodSignature(javax.lang.model.element.ExecutableElement)'
  	at org.checkerframework.common.wholeprograminference.WholeProgramInferenceScenes.updateFromLocalAssignment(WholeProgramInferenceScenes.java:234)
  	at org.checkerframework.framework.flow.CFAbstractTransfer.visitAssignment(CFAbstractTransfer.java:823)
  	at org.checkerframework.framework.flow.CFAbstractTransfer.visitAssignment(CFAbstractTransfer.java:91)
  	at org.checkerframework.dataflow.cfg.node.AssignmentNode.accept(AssignmentNode.java:62)
  	at org.checkerframework.dataflow.analysis.AbstractAnalysis.callTransferFunction(AbstractAnalysis.java:332)
  	at org.checkerframework.dataflow.analysis.ForwardAnalysisImpl.callTransferFunction(ForwardAnalysisImpl.java:373)
  	at org.checkerframework.dataflow.analysis.ForwardAnalysisImpl.performAnalysisBlock(ForwardAnalysisImpl.java:128)
  	at org.checkerframework.dataflow.analysis.ForwardAnalysisImpl.performAnalysis(ForwardAnalysisImpl.java:105)
  	at org.checkerframework.framework.flow.CFAbstractAnalysis.performAnalysis(CFAbstractAnalysis.java:109)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.analyze(GenericAnnotatedTypeFactory.java:1249)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.performFlowAnalysis(GenericAnnotatedTypeFactory.java:1161)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.checkAndPerformFlowAnalysis(GenericAnnotatedTypeFactory.java:1551)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.preProcessClassTree(GenericAnnotatedTypeFactory.java:258)
  	at org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:307)
  	at org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:165)
  	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:808)
  	at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:56)
  	at org.checkerframework.framework.source.SourceVisitor.visit(SourceVisitor.java:82)
  	at org.checkerframework.framework.source.SourceChecker.typeProcess(SourceChecker.java:894)
  	at org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:510)
  	at org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:503)
  	at org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:190)
  	at jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:828)
  	at jdk.compiler/com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:132)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1418)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1375)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:973)
  	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:311)
  	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:170)
  	at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:57)
  	at jdk.compiler/com.sun.tools.javac.Main.main(Main.java:43)
error: SourceChecker.typeProcess: unexpected Throwable (NoSuchMethodError) while processing C.java; message: 'java.lang.String scenelib.annotations.util.JVMNames.getJVMMethodSignature(javax.lang.model.element.ExecutableElement)'
  ; The Checker Framework crashed.  Please report the crash.
  Compilation unit: C.java
  Last visited tree at line 3 column 1:
  public class C {
  Exception: java.lang.NoSuchMethodError: 'java.lang.String scenelib.annotations.util.JVMNames.getJVMMethodSignature(javax.lang.model.element.ExecutableElement)'; java.lang.NoSuchMethodError: 'java.lang.String scenelib.annotations.util.JVMNames.getJVMMethodSignature(javax.lang.model.element.ExecutableElement)'
  	at org.checkerframework.common.wholeprograminference.WholeProgramInferenceScenes.updateFromLocalAssignment(WholeProgramInferenceScenes.java:234)
  	at org.checkerframework.framework.flow.CFAbstractTransfer.visitAssignment(CFAbstractTransfer.java:823)
  	at org.checkerframework.checker.initialization.InitializationTransfer.visitAssignment(InitializationTransfer.java:149)
  	at org.checkerframework.checker.initialization.InitializationTransfer.visitAssignment(InitializationTransfer.java:57)
  	at org.checkerframework.dataflow.cfg.node.AssignmentNode.accept(AssignmentNode.java:62)
  	at org.checkerframework.dataflow.analysis.AbstractAnalysis.callTransferFunction(AbstractAnalysis.java:332)
  	at org.checkerframework.dataflow.analysis.ForwardAnalysisImpl.callTransferFunction(ForwardAnalysisImpl.java:373)
  	at org.checkerframework.dataflow.analysis.ForwardAnalysisImpl.performAnalysisBlock(ForwardAnalysisImpl.java:128)
  	at org.checkerframework.dataflow.analysis.ForwardAnalysisImpl.performAnalysis(ForwardAnalysisImpl.java:105)
  	at org.checkerframework.framework.flow.CFAbstractAnalysis.performAnalysis(CFAbstractAnalysis.java:109)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.analyze(GenericAnnotatedTypeFactory.java:1249)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.performFlowAnalysis(GenericAnnotatedTypeFactory.java:1161)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.checkAndPerformFlowAnalysis(GenericAnnotatedTypeFactory.java:1551)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.preProcessClassTree(GenericAnnotatedTypeFactory.java:258)
  	at org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:307)
  	at org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:165)
  	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:808)
  	at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:56)
  	at org.checkerframework.framework.source.SourceVisitor.visit(SourceVisitor.java:82)
  	at org.checkerframework.framework.source.SourceChecker.typeProcess(SourceChecker.java:894)
  	at org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:510)
  	at org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:190)
  	at jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:828)
  	at jdk.compiler/com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:132)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1418)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1375)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:973)
  	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:311)
  	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:170)
  	at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:57)
  	at jdk.compiler/com.sun.tools.javac.Main.main(Main.java:43)
2 errors
```
This crash happens with both `-Ainfer=stubs` and `-Ainfer=jaifs`.

Here's the full output for
```
$CHECKERFRAMEWORK/checker/bin/javac C.java -processor Nullnesschecker -Ainfer=stubs -version -verbose -AprintAllQualifiers
```
and
```
$CHECKERFRAMEWORK/checker/bin-devel/javac C.java -processor Nullnesschecker -Ainfer=stubs -version -verbose -AprintAllQualifiers
```
[bin-javac-output.txt](https://github.com/typetools/checker-framework/files/4874114/bin-javac-output.txt)
[bin-devel-javac-output.txt](https://github.com/typetools/checker-framework/files/4874115/bin-devel-javac-output.txt)

--- a/framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java
+++ b/framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java
@@ -229,7 +229,11 @@ public class WholeProgramInferenceScenes implements WholeProgramInference {
 
         String className = getEnclosingClassName(lhs);
         String jaifPath = storage.getJaifPath(className);
-        AClass clazz = storage.getAClass(className, jaifPath);
+        AClass clazz =
+                storage.getAClass(
+                        className,
+                        jaifPath,
+                        (ClassSymbol) TreeUtils.elementFromDeclaration(classTree));
         ExecutableElement methodElt = TreeUtils.elementFromDeclaration(methodTree);
         AMethod method = clazz.methods.getVivify(JVMNames.getJVMMethodSignature(methodElt));
         method.setFieldsFromMethodElement(methodElt);

--- a/framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/ASceneWrapper.java
+++ b/framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/ASceneWrapper.java
@@ -163,7 +163,7 @@ public class ASceneWrapper {
     /**
      * Updates the symbol information stored in AClass for the given class. May be called multiple
      * times (and needs to be if the second parameter was null the first time it was called; only
-     * some calls provide the symbol inforamtion).
+     * some calls provide the symbol information).
      *
      * @param aClass the class representation in which the symbol information is to be updated
      * @param classSymbol the source of the symbol information; may be null, in which case this

--- a/framework/tests/whole-program-inference/non-annotated/ExpectedErrors.java
+++ b/framework/tests/whole-program-inference/non-annotated/ExpectedErrors.java
@@ -234,4 +234,11 @@ public class ExpectedErrors {
             expectsSibling1(field2);
         }
     }
+
+    class AssignParam {
+        public void f(@WholeProgramInferenceBottom Object param) {
+            // :: error: assignment.type.incompatible
+            param = ((@Top Object) null);
+        }
+    }
 }

